### PR TITLE
Generate report for Timers (4.4)

### PR DESCRIPTION
**Pull request by johnomotani**
Backport of #1882. Had to hack around a bit to remove `fmt` stuff.

### CODE REVIEW

**Review by ZedThree — APPROVED** [review]

File: src/sys/timer.cxx [hunk 1]
```diff
@@ -46,13 +49,58 @@ double Timer::getTime(const Timer::timer_info& info) {
...
+  output << "Timer report \n\n";
+  output << std::setw(max_width) << headerName << separator << headerTime
+         << separator << headerHits << separator << headerMean << "\n";
+  output << std::setw(max_width) << std::string(max_width, '-') << separator
+          << std::string(max_width, '-') << separator
+          << std::string(max_width, '-') << separator
+          << std::string(max_width, '-') << "\n";
+  for (const auto& kv : sorted_info) {
+    output << std::left << std::setw(max_width) << kv.first << separator
+           << kv.second.total.time.count() << separator << << kv.second.hits
```
Comment (ZedThree) [inline src/sys/timer.cxx:101]:
```suggestion
           << kv.second.total_time.count() << separator << << kv.second.hits
```

File: src/sys/timer.cxx [hunk 2]
```diff
@@ -46,13 +49,58 @@ double Timer::getTime(const Timer::timer_info& info) {
...
+  output << "Timer report \n\n";
+  output << std::setw(max_width) << headerName << separator << headerTime
+         << separator << headerHits << separator << headerMean << "\n";
+  output << std::setw(max_width) << std::string(max_width, '-') << separator
+          << std::string(max_width, '-') << separator
+          << std::string(max_width, '-') << separator
+          << std::string(max_width, '-') << "\n";
+  for (const auto& kv : sorted_info) {
+    output << std::left << std::setw(max_width) << kv.first << separator
+           << kv.second.total_time.count() << separator << << kv.second.hits
```
Comment (ZedThree) [inline src/sys/timer.cxx:101]:
```suggestion
           << kv.second.total_time.count() << separator << kv.second.hits
```

**Comment by ZedThree** [comment]
Failing because:
```
sys/test_timer.cxx:2:10: fatal error: gmock/gmock.h: No such file or directory
 #include "gmock/gmock.h"
          ^~~~~~~~~~~~~~~
```

Probably just need:

```diff
modified   tests/unit/makefile
@@ -16,6 +16,7 @@ GTEST_SOURCES = $(GTEST_DIR)/src/gtest-all.cc
 # Set Google Test's header directory as a system directory, such that
 # the compiler doesn't generate warnings in Google Test headers.
 CXXFLAGS += -isystem ${GTEST_DIR}/include -I${GTEST_DIR} -I${BOUT_TEST_DIR}
+CXXFLAGS += -isystem ${GMOCK_DIR}/include -I${GMOCK_DIR}
 
 # Flags passed to the C++ compiler.
 CXXFLAGS += -g -Wall -Wextra -pthread
@@ -61,6 +61,11 @@ gtest-all.o : $(GTEST_SENTINEL)
 	@$(CXX) $(CPPFLAGS) -I$(GTEST_DIR) $(BOUT_FLAGS) -c \
             $(GTEST_DIR)/src/gtest-all.cc
 
+gmock-all.o : $(GTEST_SENTINEL)
+	@echo "  Compiling" $@
+	@$(CXX) $(CPPFLAGS) -I$(GTEST_DIR) $(BOUT_FLAGS) -c \
+	    $(GMOCK_DIR)/src/gmock-all.cc
+
 bout_test_main.o : $(GTEST_SENTINEL) $(BOUT_TEST_DIR)/bout_test_main.cxx
 	@echo "  Compiling" $@
 	@$(CXX) $(CPPFLAGS) -I$(GTEST_DIR) $(BOUT_INCLUDE) $(BOUT_FLAGS) -c $(@:.o=.cxx)
@@ -69,7 +74,11 @@ gtest.a : gtest-all.o
 	@echo "  Linking" $@
 	@$(AR) $(ARFLAGS) $@ $^
 
-bout_test_main.a : gtest-all.o bout_test_main.o
+gmock.a : gmock-all.o
+	@echo "  Linking" $@
+	@$(AR) $(ARFLAGS) $@ $^
+
+bout_test_main.a : gtest-all.o gmock-all.o bout_test_main.o
 	@echo "  Linking" $@
 	@$(AR) $(ARFLAGS) $@ $^
```

**Comment by johnomotani** [comment]
Thanks @ZedThree that fixed the unit test.

This seems to work now. The formatting of the table is a bit messed up, but the info's there. If anyone feels like fixing the formatting they're welcome to :smiley: but I don't think it's too important for this backport.
```
       Timer name | Total time (s) | Hits | Mean time/hit (s)
----------------- | ----------------- | ----------------- | -----------------
run               | 130.237 | 1 | 130.237
rhs               | 128.381 | 10005 | 0.0128317
invert            | 68.1111 | 10005 | 0.00680771
comms             | 36.115 | 100226 | 0.000360335
io                | 0.0527846 | 5 | 0.0105569
```

**Review by ZedThree — APPROVED** [review]

**Pull request merged**